homeassistant/components/binary_sensor/velbus.py

```diff
@@ -4,82 +4,62 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/binary_sensor.velbus/
 """
+
 import asyncio
 import logging
 
-
-import voluptuous as vol
-
-from homeassistant.const import CONF_NAME, CONF_DEVICES
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.components.binary_sensor import PLATFORM_SCHEMA
 from homeassistant.components.velbus import DOMAIN
-import homeassistant.helpers.config_validation as cv
-
-
-DEPENDENCIES = ['velbus']
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [
-        {
-            vol.Required('module'): cv.positive_int,
-            vol.Required('channel'): cv.positive_int,
-            vol.Required(CONF_NAME): cv.string,
-            vol.Optional('is_pushbutton'): cv.boolean
-        }
-    ])
-})
+DEPENDENCIES = ['velbus']
 
 
-def setup_platform(hass, config, add_devices, discovery_info=None):
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
     """Set up Velbus binary sensors."""
     velbus = hass.data[DOMAIN]
```

Comment:
@MartinHjelmare why would I do that? I'm not getting the sensors from discovery_info object and this will always be empty, or am I missing something? Thanks for clarifying.

Reply:
Because we only want to set up the platform via discovery from the component and not via platform config. `discovery_info` will be `None` if platform config is used to set up the platform.

Reply:
It's ok to keep it like it's done now. But if you see an opportunity for simplification, go ahead.

Reply:
`discovery_info` must be serializable though.